Updates for general release of Blockstack Connect

markmhx opened the pull request:
Update the documentation widely to support the general release of Blockstack Connect and the corresponding deprecation of the Blockstack Browser.

Issues resolved:

- [ ] #560  -> needs videos and screenshot still
- [x] #566 
- [x] #565 
- [x] #564 
- [x] #553 
- [x] #552
- [x] #561

markmhx commented:
@aulneau Mind adding the video assets inline here where marked "insert video"? (https://github.com/blockstack/docs.blockstack/pull/612 and https://github.com/blockstack/docs.blockstack/pull/612)

aulneau commented:
@markmhx im pto today, will do on Monday

markmhx commented:
@hstove @agraebe @johndmulhausen Could I get your reviews on this PR today ahead of general release tomorrow?

agraebe commented:
I'll have a look

agraebe commented on _data/theFAQs.json:
```diff
@@ -151,102 +146,102 @@
          "answer": "<p>Currently, moving your data from one storage provider to another is not supported via the UI. You can do this move with assistance from Blockstack.</p>"
       },
       {
-         "category": "dappdevs",
```
The new category seems not to be visible: https://deploy-preview-612--docsproduction.netlify.app/faqs/allfaqs#dapp-developers-questions

Maybe the category id is used to filter and show these in the page?

markmhx replied:
Oops, this was the result of a sloppy search/replace on my part. I'll fix.

agraebe commented on _develop/connect/get-started.md:
````diff
@@ -162,4 +157,24 @@ if (userSession.isSignInPending()) {
\ No newline at end of file
+```
+
+## Design Guidance
````
should this maybe be a separate page? seems quite important but is getting lost in the lengthy guide content

markmhx replied:
Good idea, I'll break it out into its own page.

agraebe commented on _develop/connect/get-started.md:
```diff
@@ -162,4 +157,24 @@ if (userSession.isSignInPending()) {
+
+Many new users to your app will not be familiar with Blockstack yet and will be hesitant to click a Blockstack-branded button. Provide a generic button for users that are new to your app and Blockstack. Blockstack Connect will introduce new users to Blockstack and recognize existing users.
+
+![Design Guidance Example](./docs/call-to-action-branding.png)
```
the URL is broken

markmhx replied:
Good catch, will fix

markmhx replied:
@jasperjansz Can you provide an updated asset here that indicates the non-Messenger example app we want to highlight in the blog post as well?

agraebe commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
 * TOC
```
I find it confusing that the URL of the file doesn't line up with the title. Introduction > "overview.html".

Similar: "Guide" > "get-started.html"

markmhx replied:
Agreed, I'll modify these to align.

agraebe commented on _develop/connect/get-started.md:
```diff
@@ -162,4 +157,24 @@ if (userSession.isSignInPending()) {
+
+Blockstack is valuable to users, but it can also be a barrier to those unfamiliar with Blockstack. The following guidelines serve to remedy that and help you onboard as many new users as you can. 
+
+### Delay Blockstack onboarding as long as possible
```
should we educate them on how to refer/call the login? "Login using Secret Key" or "Login with Blockstack"?

hstove replied:
The recommendation is to just say "Get Started". I believe we had findings that "Login with X" doesn't work when you don't know what X is. Hence the introductory modal, etc

markmhx replied:
The screenshot below includes this guidance, but I'll add it to the copy as well so that it's even clearer and text-based for SEO, etc.

agraebe commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
+
+**insert video**
+
+As you can see, you simply need to add a "Get started" option to your app homepage and the library will display an introductory modal in the context of your app that explains the privacy benefits of using Blockstack and prepares users to receive a Secret Key instead of entering a traditional email and password.
```
this is a mouthful and I don't understand what a get started option in my app homepage would be

hstove replied:
Maybe this is where a short video/gif would tell the story much better.

markmhx replied:
The video should help once we've embedded it above. I'll work on the language here to make it clearer as well.

agraebe commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
+
+As you can see, you simply need to add a "Get started" option to your app homepage and the library will display an introductory modal in the context of your app that explains the privacy benefits of using Blockstack and prepares users to receive a Secret Key instead of entering a traditional email and password.
+
+The modal then triggers a popup in which Blockstack PBC's authenticator generates a 12-word mnemonic Secret Key and instructs the user to save it securely. The authenticator subsequently requests a username from the user, which will be used to identify them in your app and manage their various accounts, before routing them back to your app to start using it.
```
The first reference of a "Blockstack authenticator" - what is it? How's it related to Connect? Might need to set the context here

markmhx replied:
Will do 👍

agraebe commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
+
+The modal then triggers a popup in which Blockstack PBC's authenticator generates a 12-word mnemonic Secret Key and instructs the user to save it securely. The authenticator subsequently requests a username from the user, which will be used to identify them in your app and manage their various accounts, before routing them back to your app to start using it.
+
+This entire onboarding flow has been designed and tested to optimize the speed in which your new users start using your app with the prerequisite knowledge and credentials. 
```
Sounds like this should be in the intro on top

agraebe commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
+As you can see, you simply need to add a "Sign in" option to your homepage. The library will trigger a popup in which the authenticator prompts users to simply enter their Secret Key at which point they can resume using your app.
 
-Although [`blockstack.js`](https://github.com/blockstack/blockstack.js) exposes everything you need to handle authentication with Blockstack, there is still the hard problem of getting users familiar with the the paradigm of authentication that is privacy-first and self-sovereign. Many apps implement their own dialogues before authentication, which explain what Blockstack is and why they use it.
+At whichever point after a user has authenticated to your app, you can trigger the authenticator in a popup once more for them to a sign a transaction against any smart contract you or others have published to the Stacks blockchain.
```
This sounds like it would benefit from being another chapter just to point out that the functionality is different than sign up/in

hstove replied:
Sure, I think linking to the "Signing transactions" page would make sense here.

markmhx replied:
👍

markmhx commented on _data/navigation_home.yml:
```diff
@@ -3,7 +3,7 @@
   desc: Learn how to build decentralized applications using Blockstack's storage and authentication tools.
   icon: code
-  doc: develop/overview_auth
+  doc: browser/hello-blockstack
```
Hmm I thought we had updated the "Hello, Blockstack" with Connect but it appears that's not the case. I've reopened https://github.com/blockstack/docs.blockstack/issues/555.

markmhx commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
-## How does this compare to `blockstack.js`?
+Here's what it looks like to register new users into an app with Blockstack Connect:
+
+**insert video**
```
@jasperjansz Please provide me with a URL here for the video asset on our CDN

markmhx commented on _develop/connect/overview.md:
```diff
@@ -8,22 +8,48 @@ permalink: /:collection/:path.html
+
+Here's what users see when returning to an app to sign back in:
+
+**insert video**
```
@jasperjansz Please provide me with a URL here for the video asset on our CDN

markmhx commented:
@hstove I deleted the "Guide to Blockstack Auth" page since @johndmulhausen indicated last week that he planned to revamp or throw out the "Guides" soon anyway. As such, they weren't worth making short-term updates.

However, upon reading through that page again, I do agree that simply deleting it for this release makes the docs even more disjointed. So I'll restore it and make some updates that permit us to keep it in place for this release while avoiding confusion with the Blockstack Browser.

I also assumed we had already updated "Hello, Blockstack" with Blockstack Connect via https://github.com/blockstack/docs.blockstack/issues/555 but that appears not to be the case. So I'll make those updates as well.

I believe the sum of these updates should resolve your two concerns, but let's see upon further review once I've pushed the changes.

markmhx commented:
This initiative to update the docs for Blockstack Connect has been moved to https://github.com/blockstack/docs.blockstack/pull/616.
